src/screens/planActivity/PlanActivityScreen.tsx

```diff
@@ -98,14 +98,35 @@ export class PlanActivityScreen extends React.PureComponent<NavigationInjectedPr
     });
   };
 
+  shuffleTasks = () => {
```

Comment:
It would be nice to have some unit tests for this method. It will be hard to maintain and fix if something get broken. Also would extract the shuffleLogic to separate method, and perform in this function body only model update.